File: examples/example-ruby/lib/app/runner.rb
require_relative '../base/base_runner'
require_relative '../model/user'

module App
  class Runner < BaseRunner
    def run(user)
      user.name
    end

    def call(user = User.new)
      run(user)
    end
  end
end

def boot
  App::Runner.new
end

File: examples/example-php/src/App/Feature/Runner.php
<?php

namespace App\Feature;

use App\Base\BaseRunner;
use App\Contracts\Runnable;
use App\Model\User;

class Runner extends BaseRunner implements Runnable {
    public function run(User $user): string {
        return $user->name;
    }
}

function boot(): Runner {
    return new Runner();
}
